### PR TITLE
Support for user/password authentication

### LINKED ISSUE #45: Support for user/password authentication
### Describe the feature

Add support for user/password authentication of clients connecting to the MQTT server. The user dictionary should be read from a password file in the format:

usr1:pwd1
usr2:pwd2
usr3:pwd3

### PR DESCRIPTION
**Pull request by gmartin82**
This PR adds support for user/password authentication. User name and password information is provided via a password dictionary file.

Closes #45.

### CODE REVIEW

**Comment by gmartin82** [comment]
@JEnoch please could you review this PR?

File: zenoh-plugin-mqtt/src/lib.rs [hunk 1]
```diff
@@ -444,12 +531,30 @@ async fn handshake_v3<'a>(
...
+                MqttSessionState::new(client_id, zsession, config, handshake.sink().into());
+            Ok(handshake.ack(session, false))
+        }
+        Err(err) => {
+            log::warn!(
+                "MQTT client {} connect using v3 rejected: {}",
+                client_id,
+                err
+            );
+            Ok(handshake.bad_username_or_pwd())
```
Comment (JEnoch) [inline zenoh-plugin-mqtt/src/lib.rs:555]: `bad_username_or_pwd()` is for the case where the data in the user name or password is malformed.
See https://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718035 table 3.2.2.3.
I think `not_authorized()` is more appropriate here.

File: zenoh-plugin-mqtt/src/lib.rs [hunk 2]
```diff
@@ -544,12 +649,30 @@ async fn handshake_v5<'a>(
...
+                MqttSessionState::new(client_id, zsession, config, handshake.sink().into());
+            Ok(handshake.ack(session))
+        }
+        Err(err) => {
+            log::warn!(
+                "MQTT client {} connect using v5 rejected: {}",
+                client_id,
+                err
+            );
+            Ok(handshake.failed(ntex_mqtt::v5::codec::ConnectAckReason::BadUserNameOrPassword))
```
Comment (JEnoch) [inline zenoh-plugin-mqtt/src/lib.rs:673]: `BadUserNameOrPassword` is for the case where the data in the user name or password is malformed.
See https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901075 table 3.2.2.2.
I think `NotAuthorized` is more appropriate here.

**Review by JEnoch — CHANGES_REQUESTED** [review]

File: zenoh-plugin-mqtt/src/lib.rs (hunk 1 above)
Comment (gmartin82) [reply zenoh-plugin-mqtt/src/lib.rs]: Changed to `not_authorized()`.

File: zenoh-plugin-mqtt/src/lib.rs (hunk 2 above)
Comment (gmartin82) [reply zenoh-plugin-mqtt/src/lib.rs]: Changed to `NotAuthorized`.

**Pull request merged**